form2request/_base.py

```diff
@@ -178,6 +178,57 @@ class Request:
     headers: list[tuple[str, str]]
     body: bytes
 
+    def to_poet(self, **kwargs):
```

Comment:
That's a tricky one, but is it possible to type annotate these methods, without making the dependencies non-optional? I'm fine with merging without it.

Reply:
I have found some options, but I’m not sure about them:
- [Importing in `TYPE_CHECKING`](https://stackoverflow.com/a/61394121/939364) (1st part of the response), which would make extras required during type checking.
- [Only defining the methods when the extras are importable](https://stackoverflow.com/a/61394121/939364) (2nd part of the response), which at the very least could be confusing for users.
- [Using a TypeVar when the extra is unavailable](https://stackoverflow.com/a/72440529/939364) looks OK to my, not sure if there is any significant drawback not mentioned yet in StackOverflow.

Reply:
Regarding the first approach, with TYPE_CHECKING. 
Would it require a `requests` user to have Scrapy installed if they want to check their code (which uses to_requests), taking form2requests definitions in account?

Reply:
I think so, yes. Unintuitively, I listed options from my least favorite to my least “not favorite”.

Reply:
Option 3 does not seem to work.

```
form2request/_base.py:19: error: Invalid assignment target  [misc]
form2request/_base.py:205: error: A function returning TypeVar should receive at least one argument containing the same TypeVar  [type-var]
form2request/_base.py:219: error: Incompatible return value type (got "PreparedRequest", expected "PreparedRequest")  [return-value]
```

Reply:
Let's skip it then :)